biz.aQute.repository/src/aQute/bnd/repository/fileset/FileSetRepository.java

```diff
@@ -87,11 +92,49 @@ private BridgeRepository getBridge() throws Exception {
 				return null;
 			}
 			ResourceBuilder rb = new ResourceBuilder();
-			try (JarFile jar = new JarFile(file)) {
+			try (Jar jar = new Jar(file)) {
 				Domain manifest = Domain.domain(jar.getManifest());
 				boolean hasIdentity = rb.addManifest(manifest);
 				if (!hasIdentity) {
-					return null;
+					Optional<POM> pom = jar.getPomResources()
+						.findFirst()
+						.map(asFunction(pomResource -> new POM(null, pomResource.openInputStream(), true)));
+
+					String name;
+
+					if (jar.getModuleName() != null) {
+						name = jar.getModuleName();
+					}
+					else if (pom.isPresent()) {
+						name = pom.get()
+							.getRevision().group + ":"
+							+ pom.get()
+								.getRevision().artifact;
+					} else {
+						return null;
+					}
+
+					String version = null;
+
+					if (pom.isPresent()) {
+						version = pom.get()
+							.getRevision().version.getOSGiVersion()
+							.toString();
+					}
+					else if (jar.getModuleVersion() != null) {
+						version = new MavenVersion(jar.getModuleVersion()).getOSGiVersion()
```

Comment:
The `new MavenVersion` should be in a try/catch since the module version may not conform to maven/OSGi version rules.